movie.py

```diff
@@ -0,0 +1,178 @@
+import requests
+import csv
+from datetime import datetime, timedelta
+from collections import Counter
+
+
+class MovieDataPreparation:
+    def __init__(self, num_pages):
+        self.num_pages = num_pages
+        self.base_url = 'https://api.themoviedb.org/3'
+        self.headers = {
+            'accept': 'application/json',
+            'Authorization': 'Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIzMTI3NGFmYTRlNTUyMjRjYzRlN2Q0NmNlMTNkOTZjOSIsInN1YiI6IjVkNmZhMWZmNzdjMDFmMDAxMDU5NzQ4OSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.lbpgyXlOXwrbY0mUmP-zQpNAMCw_h-oaudAJB6Cn5c8'
+        }
+        self.movies = self.get_sample_data()
+        self.genres = self.get_genre_data()
+
+    def make_request(self, url):
+        response = requests.get(url, headers=self.headers)
+        return response.json()
+
+    def get_movie_data(self, page):
+        url = f'{self.base_url}/discover/movie?include_adult=false&include_video=false&sort_by=popularity.desc&page={page}'
+        data = self.make_request(url)
+        return data.get('results', [])
+
+    def get_genre_data(self):
+        url = f'{self.base_url}/genre/movie/list?language=en'
+        data = self.make_request(url)
+        return data.get('genres', [])
+
+    def get_sample_data(self):
+        movies = []
+        for page in range(1, self.num_pages + 1):
+            movies.extend(self.get_movie_data(page))
+        return movies
+
+    def get_all_data(self):
+        return self.movies
+
+    def get_movies_with_indexes(self):
+        movies = self.get_sample_data()
+        return movies[3:20:4]
```

Comment:
```suggestion
        return self.movies[3:20:4]
```
just use the field. you don't need to fetch the data again